### PR TITLE
Refactor to handle multiple tenants

### PR DESCRIPTION
**Pull request by saswatamcode**
This PR refactors the service to handle multiple tenants. It switches over most configuration options to flags, aside from K8s namespace name and tenant client id and secret.

It uses the `--managed-tenants` flag to look for env vars of the format `<TENANT>_CLIENT_ID` and `<TENANT>_CLIENT_SECRET` for each tenant. and uses that to initialize the obsctl config.

This also modifies the jsonnet library, to handle config for multiple tenants.

### CODE REVIEW

**Review by NautiluX — COMMENTED** [review]
/lgtm

File: main_test.go [hunk 1]
```diff
@@ -190,10 +196,10 @@ func TestInitOrReloadObsctlConfig(t *testing.T) {
 		t.Run(tc.name, func(t *testing.T) {
 			dir := t.TempDir()
 			testutil.Ok(t, os.Setenv("OBSCTL_CONFIG_PATH", path.Join(dir, "obsctl", "config.json")))
-			testutil.Ok(t, os.Setenv("MANAGED_TENANTS", tc.tenants))
 
 			// Handle test cases with pre-exisiting config.
 			if tc.prexistingConfig != nil {
+				o.managedTenants = tc.tenants
```
Comment (jessicalins) [inline main_test.go:202]: maybe we could add a test case with where `tc.tenants` has a non-expected format (e.g. not comma separated')?

Comment (saswatamcode) [reply main_test.go:202]: So if it's not comma-separated, we consider it as the name of a single tenant. There isn't really an error case here to test 🙂

Comment (jessicalins) [reply main_test.go:202]: I see - in case is then without a comma, is the input validated in any way?

Comment (jessicalins) [reply main_test.go:202]: sounds good then!

**Review by jessicalins — APPROVED** [review]
LGTM

**Pull request merged**